src/Hyperion.Benchmarks/SerializeStructsBenchmark.cs

```diff
@@ -30,8 +43,21 @@ protected override void Init()
         #endregion
 
         [Benchmark] public void Enums() => SerializeAndDeserialize(testEnum);
+        [Benchmark] public void Filtered_Enums() => SerializeAndFilteredDeserialize(testEnum);
         [Benchmark] public void Standard_Value_Types() => SerializeAndDeserialize(standardValue);
+        [Benchmark] public void Filtered_Standard_Value_Types() => SerializeAndFilteredDeserialize(standardValue);
         [Benchmark] public void Blittable_Value_Types() => SerializeAndDeserialize(blittableValue);
+        [Benchmark] public void Filtered_Blittable_Value_Types() => SerializeAndFilteredDeserialize(blittableValue);
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void SerializeAndFilteredDeserialize<T>(T elem)
+        {
+            Serializer.Serialize(elem, Stream);
```

Comment:
`Serializer` does not have types registered, it should throw an exception. How about `_filteredSerializer.Serialize(elem, Stream)`?;

Reply:
if you don't have any types registered, it automatically disable the type filtering. we don't want to force our users to always use type filtering, its an opt-in feature.

Reply:
Type filtering is only done during __deserialization__, it doesn't matter who serializes the original object, the deserializer is the one responsible for filtering the types.

Reply:
Got that - I forgot that part! :)